### DISCUSSION TITLE
Image Entity Media Sources

### DISCUSSION THREAD

**Discussion by OnFreund** [home-assistant/architecture#1029] [Entity Models] [answered]
Note: I initially [created a PR](https://github.com/home-assistant/core/pull/104743), but at @joostlek's advice, I'm starting a discussion here.

Image entities were recently added to represent static images, so that Camera entities could be dedicated to video. Some integrations have already converted their camera entities into Image entities, but doing so loses important functionality - Media Sources.
This also creates a disincentive to migrate static images from Camera to Image.

Examples of Image entities that would benefit from being a media source:
* Radar maps (e.g. Buienradar, Environment Canada, which are currently still Camera entities).
* Frigate (a custom integration) detection entities represent the last detection of say a cat in your backyard camera.

These cases, while representing a static image as opposed to a video, represent an image that is dynamic, and can change, and it would make sense for those to be media sources, so that they can be streamed/casted, etc...
I can also think of future use cases, such as an integration that fetches a random photo from google photos at a set interval, based on some search term, an integration that creates a graphical representation of other entities, etc...

While creating a dashboard with these entities and casting them is also an option, I don't believe it's the right approach because:
1. It's very cumbersome
2. The same argument could be used for any media source backed by an entity, yet we do have camera media sources.

**Comment by emontnemery** [comment, marked as answer]
The proposal to make image entities media sources is accepted, please go ahead with the PR @OnFreund 👍

**Comment by OnFreund** [comment]
Implemented in https://github.com/home-assistant/core/pull/104743
